flink-table/flink-table-planner/src/main/scala/org/apache/flink/table/api/internal/TableEnvImpl.scala

```diff
@@ -532,7 +542,13 @@ abstract class TableEnvImpl(
 
       case Some(tableSink) =>
         // validate schema of source table and table sink
-        TableSinkUtils.validateSink(table.getQueryOperation, sinkTablePath.asJava, tableSink)
+        TableSinkUtils.validateSink(insertOptions.staticPartitions, table.getQueryOperation,
+          sinkTablePath.asJava, tableSink)
+        tableSink match {
+          case partitionableSink: PartitionableTableSink =>
```

Comment:
Only set static partitions when sink has partitioned fields, i.e. is a partitioned sink. Currently, we set a partition map always which may have unexpected effects.